rocketpy/motors/liquid_motor.py

```diff
@@ -264,7 +273,34 @@ def exhaust_velocity(self):
         mass flow rate. Therefore, this will vary with time if the mass flow
         rate varies with time.
         """
-        return self.thrust / (-1 * self.mass_flow_rate)
+        if not isinstance(self.thrust.source, np.ndarray):
+            return self.thrust / (-1 * self.mass_flow_rate)
+
+        times, thrusts = self.thrust.source[:, 0], self.thrust.source[:, 1]
+        mass_flow_rates = self.mass_flow_rate(times)
+
+        # Avoid division by zero by replacing 0 with NaN
+        mass_flow_rates_nonzero = np.where(
+            np.isclose(mass_flow_rates, 0), np.nan, mass_flow_rates
+        )
+        ext_vel = -thrusts / mass_flow_rates_nonzero
+
+        if np.any(np.isnan(ext_vel)):
+            warnings.warn(
+                "Exhaust velocity is NaN for some values of time, probably due to a"
+                "division by zero. Check if the mass flow rate curve isn't zero for "
+                "some values of time. The times used to calculate the exhaust "
+                "velocity are the same as the computed thrust curve."
+            )
+
```

Comment:
So the user should never input a mass flow rate curve with any point that is 0?

Does exhaust velocity having NaN values break anything else in the code? Seems like it would make some plots break at least

Reply:
When there's a NaN in the datasource, matplotlib automatically ignore the plot in that point, which is a good behavior.

A problem happens when we are taking the average exhaust velocity, which is currently returning Nan.

The mas flow rate should never be zero if the thrust is different than zero, it doesn't make sense.

Reply:
> The mas flow rate should never be zero if the thrust is different than zero, it doesn't make sense.

And if both the mass flow rate and the thrust is zero? Exhaust velocity should be zero, is that what happens?





> A problem happens when we are taking the average exhaust velocity, which is currently returning Nan.

I guess we should skip over the NaN point like matplotlib does here?

Reply:
if thrust is zero in a certain point and the mass flow rate is zero as well, the code will throw this warning. I agree this should be calculated as zero at this point, but I don't see an easy solution for this, I need to think about it.

There's this option of calculating the time boundaries for the exhaust velocity instead of simply using the same time points from the thrust source. That should be just a bit more complicated, but a better solution. Lemme think a bit.

Reply:
I came this a cleaner version of the code, hope you liked it. It is working flawlessly now.

![image](https://github.com/RocketPy-Team/RocketPy/assets/63590233/97f1c12c-60b4-483a-9814-d5c8b2e97a44)